packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -172,6 +175,48 @@ export interface LightningElement extends HTMLElementTheGoodParts, AccessibleEle
     errorCallback?(error: any, stack: string): void;
 }
 
+function setShadowRenderMode(vm: VM) {
+    const {
+        def: { ctor },
+    } = vm;
+    const { preferNativeShadow, renderMode } = ctor;
+    const isNativeShadowRootDefined = globalThis['$isNativeShadowRootDefined$'];
+
+    if (process.env.NODE_ENV !== 'production') {
+        if (!isUndefined(renderMode)) {
+            assert.invariant(
+                renderMode === 'light' || renderMode === 'shadow',
+                `Invalid value for static property renderMode: '${renderMode}'. renderMode must be either 'light' or 'shadow'.`
+            );
+        }
+
+        if (!isUndefined(preferNativeShadow)) {
+            assert.invariant(
+                typeof preferNativeShadow === 'boolean',
+                `Invalid value for static property preferNativeShadow: '${preferNativeShadow}'. preferNativeShadow must be a boolean value.`
+            );
+        }
+    }
+
+    if (!features.ENABLE_LIGHT_DOM_COMPONENTS) {
+        assert.isTrue(
+            renderMode !== 'light',
+            `${
+                ctor.name || 'Anonymous class'
+            } is an invalid LWC component. Light DOM components are not available in this environment.`
+        );
+    }
+
+    if (renderMode === 'light') {
+        vm.shadowRenderMode = ShadowRenderMode.None;
+    } else if (isTrue(preferNativeShadow) && isNativeShadowRootDefined) {
+        vm.shadowRenderMode = ShadowRenderMode.SyntheticNative;
+    } else {
+        // Default to synthetic mode.
+        vm.shadowRenderMode = ShadowRenderMode.Synthetic;
```

Comment:
It is strange that the default render mode synthetic. When the browser support shadow DOM and synthetic shadow DOM polyfill (which is what I consider the default) the shadow render mode is still set to synthetic.